### PR TITLE
Clean docstrings in repo

### PR DESCRIPTION
**Pull request by ninamiolane**
This PR cleans docstrings and specifically:
- Use singular form for point and tangent vectors, see issue #505.
- Update convention to write optional parameters, which is now: `Optional, default: default_value.` on a new line of the parameter description.
- Update contributing.rst to add the convention for optional parameters in docstrings.
- Add missing docstrings.

`landmarks.py` and `discretized_curves.py` remain to be done @nguigs .

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -461,6 +460,7 @@ Here's a generic docstring template::
          Write a short description of parameter my_param_1.
       my_param_2 : str, {'vector', 'matrix'}
          Write a short description of parameter my_param_2.
+         Optional, default: 'vector'.
```
Comment (nguigs) [inline docs/contributing.rst:463]: what about the convention that when listing the possible values of an optional parameter, the default is the first one to appear, that is, {'vector', 'matrix'}, optional implies default: 'vector'?

File: geomstats/geometry/beta_distributions.py [hunk 2]
```diff
@@ -19,6 +19,13 @@ class BetaDistributions(EmbeddedManifold):
     r"""Class for the manifold of beta distributions.
 
     This is :math: Beta = `R_+^* \times R_+^*`.
+
+    Attributes
+    ----------
+    dim : int
+        Dimension of the manifold of beta distributions.
```
Comment (nguigs) [inline geomstats/geometry/beta_distributions.py:26]: should we write explicitely that the dimension is 2?

File: geomstats/geometry/beta_distributions.py [hunk 3]
```diff
@@ -19,6 +19,13 @@ class BetaDistributions(EmbeddedManifold):
...
+    embedding_manifold : Manifold
+        Embedding manifold.
```
Comment (nguigs) [inline geomstats/geometry/beta_distributions.py:28]: Actually, as per your response in an earlier PR, as the beta manifold is an open set of R^2 it should not be considered an embedded manifold of R^2

File: geomstats/geometry/beta_distributions.py [hunk 4]
```diff
@@ -221,12 +237,17 @@ def exp(self, tangent_vec, base_point, n_steps=N_STEPS):
         Parameters
         ----------
         tangent_vec : array-like, shape=[..., dim]
+            Tangent vector.
         base_point : array-like, shape=[..., dim]
+            Base point.
         n_steps : int
+            Number of steps for integration.
+            Optional, default: N_STEPS.
```
Comment (nguigs) [inline geomstats/geometry/beta_distributions.py:245]: maybe write the value of N_STEPS otherwise it won't appear in the doc but only in the code?

File: geomstats/geometry/connection.py [hunk 5]
```diff
@@ -14,12 +14,18 @@
...
+    r"""Class for affine connections.
 
     Parameters
     ----------
-    dim: int
+    dim : int
         Dimension of the underlying manifold.
+    default_point_type : str, {\'vector\', \'matrix\'}
+        Point type.
+        Optional, default: \'vector\'.
```
Comment (nguigs) [inline geomstats/geometry/connection.py:25]: ditto

File: geomstats/geometry/euclidean.py [hunk 6]
```diff
@@ -40,14 +46,17 @@ def random_uniform(self, n_samples=1, bound=1.):
 
         Parameters
         ----------
-        n_samples : int, optional
-            default: 1
-        bound : float, optional
-            default: 1.0
+        n_samples : int
+            Number of samples.
+            Optional, default: 1.
```
Comment (nguigs) [inline geomstats/geometry/euclidean.py:51]: no dot if its an integer -> 1

File: geomstats/geometry/euclidean.py [hunk 7]
```diff
@@ -92,12 +109,14 @@ def exp(self, tangent_vec, base_point):
         Parameters
         ----------
         tangent_vec : array-like, shape=[..., dim]
-
+            Tangent vector.
```
Comment (nguigs) [inline geomstats/geometry/euclidean.py:112]: what about adding "at the base point"?

File: geomstats/geometry/general_linear.py [hunk 8]
```diff
@@ -94,8 +120,10 @@ def exp(cls, tangent_vec, base_point=None):
         Parameters
         ----------
         tangent_vec : array-like, shape=[..., n, n]
+            Tangent vector.
         base_point : array-like, shape=[..., n, n]}
```
Comment (nguigs) [inline geomstats/geometry/general_linear.py:124]: typo: extra }

File: geomstats/geometry/general_linear.py [hunk 9]
```diff
@@ -111,16 +139,18 @@ def exp(cls, tangent_vec, base_point=None):
...
         algebra element if it is not the identity.
 
         Parameters
         ----------
         point : array-like, shape=[..., n, n]
+            Point.
         base_point : array-like, shape=[..., n, n]
-            Defaults to identity.
+            Base point.
+            Optional, default to identity if None.
```
Comment (nguigs) [inline geomstats/geometry/general_linear.py:153]: I think this should be either default: identity or defaults to identity (with the s)

File: geomstats/geometry/grassmannian.py [hunk 10]
```diff
@@ -52,12 +59,14 @@ def belongs(self, point, tolerance=TOLERANCE):
         Parameters
         ----------
         point : array-like, shape=[..., n, n]
+            Point to be checked.
         tolerance : int
-            default: TOLERANCE
+            Optional, default: TOLERANCE
```
Comment (nguigs) [inline geomstats/geometry/grassmannian.py:64]: idem, this way it won't be possible what the default tol is from the doc

File: geomstats/geometry/hyperbolic.py [hunk 11]
```diff
@@ -362,10 +366,12 @@ def random_uniform(self, n_samples=1, bound=1.):
 
         Parameters
         ----------
-        n_samples : int, optional
+        n_samples : int
             Number of samples.
-        bound: float, optional
+            Optional, default: 1.
```
Comment (nguigs) [inline geomstats/geometry/hyperbolic.py:371]: 1

File: geomstats/geometry/hyperboloid.py [hunk 12]
```diff
@@ -51,11 +51,13 @@ class Hyperboloid(Hyperbolic, EmbeddedManifold):
...
         Dimension of the hyperbolic space.
-    point_type : str, {'extrinsic', 'intrinsic'}, optional
+    coords_type : str, {'extrinsic', 'intrinsic'}
         Default coordinates to represent points in hyperbolic space.
-    scale : int, optional
+        Optional, default: 'extrinsic'.
+    scale : int
         Scale of the hyperbolic space, defined as the set of points
         in Minkowski space whose squared norm is equal to -scale.
+        Optional, default: 1.
```
Comment (nguigs) [inline geomstats/geometry/hyperboloid.py:60]: 1

File: geomstats/geometry/product_manifold.py [hunk 13]
```diff
@@ -80,10 +84,11 @@ def belongs(self, point, point_type=None):
             Point.
         point_type : str, {'vector', 'matrix'}
             Representation of point.
+            Optional, default: None.
 
         Returns
         -------
-        belongs : array-like, shape=[..., 1]
+        belongs : array-like, shape=[...,]
             Array of booleans evaluating if the corresponding points
```
Comment (nguigs) [inline geomstats/geometry/product_manifold.py:92]: plural form here --> should be singular?

File: geomstats/geometry/spd_matrices.py [hunk 14]
```diff
@@ -24,15 +30,41 @@ def __init__(self, n):
...
+        mat : array-like, shape=[..., n, n]
+            Matrix to be checked.
+        atol : float
+            Tolerance.
+            Optional, default: TOLERANCE.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean denoting if mat is a SPD matrix.
```
Comment (nguigs) [inline geomstats/geometry/spd_matrices.py:46]: typo: is an SPD

File: geomstats/geometry/symmetric_matrices.py [hunk 15]
```diff
@@ -42,7 +59,18 @@ def get_basis(self):
     @staticmethod
     @geomstats.vectorization.decorator(['matrix'])
     def to_vector(mat):
-        """Convert the symmetric part of a symmetric matrix into a vector."""
+        """Convert the symmetric part of a symmetric matrix into a vector.
```
Comment (nguigs) [inline geomstats/geometry/symmetric_matrices.py:62]: Redudant: convert the symmetric matrix or the symmetric part of a matrix? As is_symmetric is checked, I suggest the former

File: geomstats/learning/exponential_barycenter.py [hunk 16]
```diff
@@ -24,21 +24,21 @@ def _default_gradient_descent(
         Input points lying in the Lie Group.
     weights : array-like, shape=[n_samples,]
```
Comment (nguigs) [inline geomstats/learning/exponential_barycenter.py:25]: Do we keep n_samples here?

File: geomstats/learning/exponential_barycenter.py [hunk 17]
```diff
@@ -24,21 +24,21 @@ def _default_gradient_descent(
...
-        The maximum number of iterations to perform in the gradient descent.
+        Maximum number of iterations to perform in the gradient descent.
     epsilon : float, optional (defaults to 1e-6)
-        The tolerance to reach convergence. The exstrinsic norm of the
+        Tolerance to reach convergence. The exstrinsic norm of the
         gradient is used as criterion.
     step : float, optional (defaults to 1.)
-        The learning rate in the gradient descent.
+        Learning rate in the gradient descent.
     verbose : bool
```
Comment (nguigs) [inline geomstats/learning/exponential_barycenter.py:35]: Default: ...
Also missing default value for verbose

File: geomstats/learning/exponential_barycenter.py [hunk 18]
```diff
@@ -91,14 +91,14 @@ class ExponentialBarycenter(BaseEstimator):
...
         Level of verbosity to inform about convergence.
```
Comment (nguigs) [inline geomstats/learning/exponential_barycenter.py:103]: idem

File: geomstats/learning/kmeans.py [hunk 19]
```diff
@@ -19,19 +19,15 @@ class RiemannianKMeans(TransformerMixin, ClusterMixin, BaseEstimator):
...
         How to initialize centroids at the beginning of the algorithm. The
         choice 'random' will select training points as initial centroids
         uniformly at random.
-
     tol : float
         Convergence factor. Convergence is achieved when the difference of mean
         distance between two steps is lower than tol.
-
     verbose : int
         If verbose > 0, information will be printed during learning.
```
Comment (nguigs) [inline geomstats/learning/kmeans.py:32]: no default values here?

File: geomstats/learning/kmeans.py [hunk 20]
```diff
@@ -65,9 +61,8 @@ def fit(self, X, max_iter=100):
         X : array-like, shape=[n_samples, n_features]
             Training data, where n_samples is the number of samples and
             n_features is the number of features.
-
         max_iter : int
-            Maximum number of iterations
+            Maximum number of iterations.
```
Comment (nguigs) [inline geomstats/learning/kmeans.py:65]: optional

File: geomstats/learning/pca.py [hunk 21]
```diff
@@ -91,12 +91,21 @@ def _infer_dimension_(spectrum, n_samples, n_features):
...
     of the Frechet mean.
+
+    Parameters
+    ----------
+    metric : RiemannianMetric
+        Riemannian metric.
+    n_components : int
+        Number of principal components.
+    point_type : str, {\'vector\', \'matrix\'}
+        Point type.
```
Comment (nguigs) [inline geomstats/learning/pca.py:108]: point_type is optional and by default it is metric.default_point_type

Comment (nguigs) [reply geomstats/learning/pca.py:108]: in fact it should be removed..

File: tests/test_beta_distributions.py [hunk 22]
```diff
@@ -122,12 +122,3 @@ def test_christoffels_vectorization(self):
...
         self.assertAllClose(result, expected)
-
-    def test_inner_product_matrix(self):
-        point = gs.array([1., 1.])
-        result = self.beta.metric.inner_product_matrix(point)
-        expected = gs.array([[1., -0.644934066], [-0.644934066, 1.]])
-        self.assertAllClose(result, expected)
-
-        self.assertRaises(
-            ValueError, self.metric.inner_product_matrix)
```
Comment (nguigs) [inline tests/test_beta_distributions.py:133]: Why did you remove this test? Maybe a rebase error

File: geomstats/geometry/hypersphere.py [hunk 23]
```diff
@@ -62,15 +62,15 @@ def belongs(self, point, tolerance=TOLERANCE):
         Parameters
         ----------
         point : array-like, shape=[..., dim + 1]
-            Points in Euclidean space.
-        tolerance : float, optional
-            Tolerance at which to evaluate norm == 1 (default: TOLERANCE).
+            Point in Euclidean space.
+        tolerance : floatl
```
Comment (nguigs) [inline geomstats/geometry/hypersphere.py:66]: typo: floatl -> float

File: geomstats/geometry/special_orthogonal.py [hunk 24]
```diff
@@ -43,30 +43,64 @@ def __init__(self, n):
...
 
     @classmethod
     def inverse(cls, point):
-        """Return the transpose matrix of point."""
+        """Return the transpose matrix of point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n, n]
+            Point in SO(n)
```
Comment (nguigs) [inline geomstats/geometry/special_orthogonal.py:68]: full stop

File: geomstats/geometry/special_orthogonal.py [hunk 25]
```diff
@@ -359,10 +402,12 @@ def skew_matrix_from_vector(self, vec):
         Parameters
         ----------
         vec : array-like, shape=[..., dim]
+            Vector.
 
         Returns
         -------
         skew_mat : array-like, shape=[..., n, n]
+            Skew symmetric matrix.
```
Comment (nguigs) [inline geomstats/geometry/special_orthogonal.py:410]: skew-symmetric?

File: geomstats/geometry/embedded_manifold.py [hunk 26]
```diff
@@ -28,6 +34,11 @@ def intrinsic_to_extrinsic_coords(self, point_intrinsic):
         ----------
         point_intrinsic : array-like, shape=[..., dim]
             Point in the embedded manifold in intrinsic coordinates.
+
+        Returns
+        -------
+        point_extrinsic : array-like, shape=[..., dim_embedding]
```
Comment (nguigs) [inline geomstats/geometry/embedded_manifold.py:40]: what about `ambiant_dim`?

File: geomstats/geometry/beta_distributions.py (hunk 3 above)
Comment (ninamiolane) [reply geomstats/geometry/beta_distributions.py:28]: Hm, let's re-discuss this? In this case, I would say that the quadrant [0, \infinity]^2 can be seen as an embedded manifold in R^2.

File: geomstats/geometry/beta_distributions.py (hunk 4 above)
Comment (ninamiolane) [reply geomstats/geometry/beta_distributions.py]: Yes, I hesitated on whether to put the actual value too. The risk with the actual value is that we might forget to change the docstring when we update the value of N_STEPS.

I have changed it though, let's try to think about it.

File: geomstats/geometry/connection.py (hunk 5 above)
Comment (ninamiolane) [reply geomstats/geometry/connection.py:25]: I'd rather stay with the more explicit convention if that's ok?

File: geomstats/geometry/embedded_manifold.py (hunk 26 above)
Comment (ninamiolane) [reply geomstats/geometry/embedded_manifold.py:40]: We make the distinction between embedding manifold and ambient manifold. Ambient manifold is the manifold to which landmarks or curves belong in Landmarks and DiscreteCurves classes: it is not an embedding manifold.

Here, we really want the embedding manifold with its dimension.

File: geomstats/geometry/euclidean.py (hunk 6 above)
Comment (ninamiolane) [reply geomstats/geometry/euclidean.py:51]: Yes, but I think we want a full stop at the end of the description line?

File: geomstats/geometry/euclidean.py (hunk 7 above)
Comment (ninamiolane) [reply geomstats/geometry/euclidean.py]: Done, in this file and all files of `geometry`.

File: geomstats/geometry/general_linear.py (hunk 8 above)
Comment (ninamiolane) [reply geomstats/geometry/general_linear.py]: Good eye!

File: geomstats/geometry/general_linear.py (hunk 9 above)
Comment (ninamiolane) [reply geomstats/geometry/general_linear.py]: I checked with a native, `defaults to` works, I have updated all the files.

File: geomstats/geometry/grassmannian.py (hunk 10 above)
Comment (ninamiolane) [reply geomstats/geometry/grassmannian.py]: Done.

File: geomstats/geometry/hyperbolic.py (hunk 11 above)
Comment (ninamiolane) [reply geomstats/geometry/hyperbolic.py:371]: Let's keep the full stop at the end of the sentence?

File: geomstats/geometry/hypersphere.py (hunk 23 above)
Comment (ninamiolane) [reply geomstats/geometry/hypersphere.py]: Done.

File: geomstats/geometry/product_manifold.py (hunk 13 above)
Comment (ninamiolane) [reply geomstats/geometry/product_manifold.py]: Done.

File: geomstats/geometry/spd_matrices.py (hunk 14 above)
Comment (ninamiolane) [reply geomstats/geometry/spd_matrices.py]: Done.

File: geomstats/geometry/special_orthogonal.py (hunk 24 above)
Comment (ninamiolane) [reply geomstats/geometry/special_orthogonal.py]: Done.

File: geomstats/geometry/special_orthogonal.py (hunk 25 above)
Comment (ninamiolane) [reply geomstats/geometry/special_orthogonal.py]: Done.

File: geomstats/geometry/symmetric_matrices.py (hunk 15 above)
Comment (ninamiolane) [reply geomstats/geometry/symmetric_matrices.py]: Done.

File: geomstats/learning/exponential_barycenter.py (hunk 16 above)
Comment (ninamiolane) [reply geomstats/learning/exponential_barycenter.py]: Removed.

File: geomstats/learning/exponential_barycenter.py (hunk 17 above)
Comment (ninamiolane) [reply geomstats/learning/exponential_barycenter.py]: Done.

File: geomstats/learning/exponential_barycenter.py (hunk 18 above)
Comment (ninamiolane) [reply geomstats/learning/exponential_barycenter.py]: Done.

File: geomstats/learning/kmeans.py (hunk 19 above)
Comment (ninamiolane) [reply geomstats/learning/kmeans.py]: Added.

File: geomstats/learning/kmeans.py (hunk 20 above)
Comment (ninamiolane) [reply geomstats/learning/kmeans.py:69]: Done.

File: geomstats/learning/pca.py (hunk 21 above)
Comment (ninamiolane) [reply geomstats/learning/pca.py]: Why?

File: tests/test_beta_distributions.py (hunk 22 above)
Comment (ninamiolane) [reply tests/test_beta_distributions.py:133]: No, this function does not raise an error for this case anymore, as it was not consistent with the behavior of other methods in the code.

Comment (nguigs) [reply tests/test_beta_distributions.py:133]: Ok, that’s better, but why not remove just the last assertRaises and keep the rest of the test?

File: geomstats/geometry/embedded_manifold.py (hunk 26 above)
Comment (nguigs) [reply geomstats/geometry/embedded_manifold.py:40]: Ok

File: tests/test_beta_distributions.py (hunk 22 above)
Comment (ninamiolane) [reply tests/test_beta_distributions.py:133]: Ok!
